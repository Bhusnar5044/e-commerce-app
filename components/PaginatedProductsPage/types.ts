import { Product } from '@utils/types';

export interface IPageProps {
    products: Product[];
    currentPage: number;
    totalProducts: number;
    perPage: number;
}
