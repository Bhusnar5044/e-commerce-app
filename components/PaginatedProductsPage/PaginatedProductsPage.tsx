import { FC, memo } from 'react';
import Pagination from '../Pagination';
import ProductList from '../ProductList';
import { IPageProps } from './types';

export const PaginatedProductsPage: FC<IPageProps> = memo(({ currentPage, totalProducts, perPage, products }) => {
    return (
        <div>
            <ProductList title="Featured Products" items={products ?? []} />
            <Pagination
                totalItems={totalProducts}
                currentPage={currentPage}
                itemsPerPage={perPage}
                renderPageLink={(page) => `/${page}`}
            />
        </div>
    );
});

PaginatedProductsPage.displayName = 'PaginatedProductsPage';
