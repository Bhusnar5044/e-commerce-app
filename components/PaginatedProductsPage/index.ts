export { PaginatedProductsPage as default } from './PaginatedProductsPage';
