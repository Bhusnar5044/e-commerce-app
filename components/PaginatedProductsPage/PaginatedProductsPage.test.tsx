import { renderWithStore } from '@test-utils';
import ProductsMock from '@utils/testData/productList.json';
import { PaginatedProductsPage } from './PaginatedProductsPage';

describe('PaginatedProductsPage', () => {
    it('should render properly', () => {
        const { container } = renderWithStore(
            <PaginatedProductsPage products={ProductsMock} currentPage={1} totalProducts={1} perPage={1} />,
        );
        expect(container).toMatchSnapshot();
    });
});
