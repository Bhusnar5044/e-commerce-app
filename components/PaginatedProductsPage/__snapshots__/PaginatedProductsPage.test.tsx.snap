// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginatedProductsPage should render properly 1`] = `
<div>
  <div>
    <div
      class="space-y-4"
    >
      <div
        class="flex justify-between items-center sticky top-[4rem] z-30 bg-white"
      >
        <h3
          class="font-bold text-3xl"
        >
          Featured Products
        </h3>
        <div
          class="flex gap-2 items-center"
        >
          <label
            class="mb-2 text-sm font-medium text-gray-900 sr-only dark:text-white"
            for="default-search"
          >
            Search
          </label>
          <div
            class="relative"
          >
            <input
              class="block w-full p-2 pl-4 text-sm text-gray-900 border border-gray-300 rounded-lg focus:ring-blue-500 focus:border-blue-500  dark:border-gray-600 dark:placeholder-gray-400 dark:focus:ring-blue-500 dark:focus:border-blue-500"
              id="default-search"
              placeholder="Search Mockups, Logos..."
              required=""
              type="search"
              value=""
            />
            <div
              class="absolute inset-y-1 right-0 flex items-center pr-3 pointer-events-cursor"
            >
              <svg
                aria-hidden="true"
                class="w-4 h-4 text-gray-500 dark:text-gray-400"
                fill="none"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m19 19-4-4m0-7A7 7 0 1 1 1 8a7 7 0 0 1 14 0Z"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
          <div
            class="flex items-center justify-center py-2"
          >
            <div
              class="relative inline-block text-left"
            >
              <span
                class="rounded-md shadow-sm"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="menu"
                  class="inline-flex justify-center w-full px-4 py-2 text-sm font-medium leading-5 text-gray-700 transition duration-150 ease-in-out bg-white border border-gray-300 rounded-md hover:text-gray-500 focus:outline-none focus:border-blue-300 focus:shadow-outline-blue "
                  data-headlessui-state=""
                  id="headlessui-menu-button-:r0:"
                  type="button"
                >
                  <span>
                    Sort
                  </span>
                  <svg
                    class="w-5 h-5 ml-2 -mr-1"
                    fill="currentColor"
                    viewBox="0 0 20 20"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4"
      >
        <div
          class="bg-white group cursor-pointer rounded-xl border p-3 space-y-4"
        >
          <div
            class="aspect-square rounded-xl bg-gray-100 relative"
          >
            <img
              alt=""
              class="aspect-square object-cover rounded-md"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
            <div
              class="opacity-0 group-hover:opacity-100 transition absolute w-full px-6 bottom-5"
            >
              <div
                class="flex gap-x-6 justify-center"
              >
                <button
                  class="rounded-full flex items-center justify-center bg-white border shadow-md p-2 hover:scale-110 transition"
                  title="iconButton"
                >
                  <svg
                    class="text-gray-600"
                    fill="none"
                    height="20"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m21 21-6-6m6 6v-4.8m0 4.8h-4.8"
                    />
                    <path
                      d="M3 16.2V21m0 0h4.8M3 21l6-6"
                    />
                    <path
                      d="M21 7.8V3m0 0h-4.8M21 3l-6 6"
                    />
                    <path
                      d="M3 7.8V3m0 0h4.8M3 3l6 6"
                    />
                  </svg>
                </button>
                <button
                  class="rounded-full flex items-center justify-center bg-white border shadow-md p-2 hover:scale-110 transition"
                  title="iconButton"
                >
                  <svg
                    class="text-gray-600"
                    fill="none"
                    height="20"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="8"
                      cy="21"
                      r="1"
                    />
                    <circle
                      cx="19"
                      cy="21"
                      r="1"
                    />
                    <path
                      d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div>
            <p
              class="font-semibold text-lg truncate"
            >
              iPhone 9
            </p>
            <p
              class="text-sm text-gray-500"
            >
              smartphones
            </p>
          </div>
          <div
            class="flex items-center justify-between"
          >
            <div
              class="font-semibold"
            >
              $549.00
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex items-center justify-center my-8"
    >
      <a
        class="text-success-dark px-4 py-2 mx-1 rounded-full text-sm font-semibold no-underline"
        href="/1"
      >
        1
      </a>
    </div>
  </div>
</div>
`;
